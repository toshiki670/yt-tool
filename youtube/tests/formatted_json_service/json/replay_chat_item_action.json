{
  "replayChatItemAction": {
    "actions": [
      {
        "clickTrackingParams": "MASKED_CLICK_TRACKING_PARAMS",
        "addChatItemAction": {
          "item": {
            "liveChatPaidMessageRenderer": {
              "id": "MASKED_ID",
              "timestampUsec": "1744042617450013",
              "authorName": {
                "simpleText": "Masked Author Name"
              },
              "authorPhoto": {
                "thumbnails": [
                  {
                    "url": "https://yt4.ggpht.com/MASKED_URL1=s32-c-k-c0x00ffffff-no-rj",
                    "width": 32,
                    "height": 32
                  },
                  {
                    "url": "https://yt4.ggpht.com/MASKED_URL1=s64-c-k-c0x00ffffff-no-rj",
                    "width": 64,
                    "height": 64
                  }
                ]
              },
              "purchaseAmountText": {
                "simpleText": "¥1,000"
              },
              "message": {
                "runs": [
                  {
                    "text": "Masked message content"
                  },
                  {
                    "emoji": {
                      "emojiId": "❗",
                      "shortcuts": [
                        ":red_exclamation_mark:",
                        ":exclamation:",
                        ":heavy_exclamation_mark:"
                      ],
                      "searchTerms": ["red", "exclamation", "mark", "heavy"],
                      "image": {
                        "thumbnails": [
                          {
                            "url": "https://fonts.gstatic.com/s/e/notoemoji/15.1/2757/72.png"
                          }
                        ],
                        "accessibility": {
                          "accessibilityData": {
                            "label": "❗"
                          }
                        }
                      }
                    }
                  }
                ]
              },
              "headerBackgroundColor": 4294947584,
              "headerTextColor": 3741319168,
              "bodyBackgroundColor": 4294953512,
              "bodyTextColor": 3741319168,
              "authorExternalChannelId": "MASKED_CHANNEL_ID",
              "authorNameTextColor": 2315255808,
              "contextMenuEndpoint": {
                "clickTrackingParams": "MASKED_CLICK_TRACKING_PARAMS",
                "commandMetadata": {
                  "webCommandMetadata": {
                    "ignoreNavigation": true
                  }
                },
                "liveChatItemContextMenuEndpoint": {
                  "params": "MASKED_PARAMS"
                }
              },
              "timestampColor": 2147483648,
              "contextMenuAccessibility": {
                "accessibilityData": {
                  "label": "Chat actions"
                }
              },
              "trackingParams": "MASKED_TRACKING_PARAMS",
              "authorBadges": [
                {
                  "liveChatAuthorBadgeRenderer": {
                    "customThumbnail": {
                      "thumbnails": [
                        {
                          "url": "https://yt3.ggpht.com/MASKED_URL2=s16-c-k",
                          "width": 16,
                          "height": 16
                        },
                        {
                          "url": "https://yt3.ggpht.com/MASKED_URL2=s32-c-k",
                          "width": 32,
                          "height": 32
                        }
                      ]
                    },
                    "tooltip": "Member (6 months)",
                    "accessibility": {
                      "accessibilityData": {
                        "label": "Member (6 months)"
                      }
                    }
                  }
                }
              ],
              "textInputBackgroundColor": 822083583,
              "creatorHeartButton": {
                "creatorHeartViewModel": {
                  "creatorThumbnail": {
                    "sources": [
                      {
                        "url": "https://yt3.ggpht.com/MASKED_URL3=s48-c-k-c0x00ffffff-no-rj"
                      }
                    ]
                  },
                  "heartedIcon": {
                    "sources": [
                      {
                        "clientResource": {
                          "imageName": "full_heart-filled"
                        }
                      }
                    ]
                  },
                  "unheartedIcon": {
                    "sources": [
                      {
                        "clientResource": {
                          "imageName": "full_heart"
                        }
                      }
                    ],
                    "processor": {
                      "borderImageProcessor": {
                        "imageTint": {
                          "color": 3741319168
                        }
                      }
                    }
                  },
                  "heartedHoverText": "❤ by Masked Channel",
                  "heartedAccessibilityLabel": "Remove heart",
                  "unheartedAccessibilityLabel": "Heart",
                  "engagementStateKey": "MASKED_ENGAGEMENT_STATE_KEY",
                  "loggingDirectives": {
                    "trackingParams": "MASKED_TRACKING_PARAMS",
                    "visibility": {
                      "types": "4"
                    }
                  }
                }
              },
              "isV2Style": true,
              "replyButton": {
                "pdgReplyButtonViewModel": {
                  "replyButton": {
                    "buttonViewModel": {
                      "iconName": "CHAT",
                      "onTap": {
                        "innertubeCommand": {
                          "clickTrackingParams": "MASKED_CLICK_TRACKING_PARAMS",
                          "showEngagementPanelEndpoint": {
                            "identifier": {
                              "surface": "ENGAGEMENT_PANEL_SURFACE_LIVE_CHAT",
                              "tag": "PAreply_thread"
                            },
                            "globalConfiguration": {
                              "params": "MASKED_PARAMS"
                            },
                            "engagementPanelPresentationConfigs": {
                              "engagementPanelPopupPresentationConfig": {
                                "popupType": "PANEL_POPUP_TYPE_DIALOG"
                              }
                            }
                          }
                        }
                      },
                      "accessibilityText": "Reply",
                      "style": "BUTTON_VIEW_MODEL_STYLE_CUSTOM",
                      "trackingParams": "MASKED_TRACKING_PARAMS",
                      "type": "BUTTON_VIEW_MODEL_TYPE_TONAL",
                      "buttonSize": "BUTTON_VIEW_MODEL_SIZE_XSMALL",
                      "customBackgroundColor": 218103808,
                      "customFontColor": 3741319168,
                      "onVisible": {
                        "innertubeCommand": {
                          "clickTrackingParams": "MASKED_CLICK_TRACKING_PARAMS",
                          "logFlowLoggingEventCommand": {
                            "flowEventType": 1,
                            "flowEventNamespace": "FLOW_EVENT_NAMESPACE_LIVE_CHAT_ENGAGEMENT",
                            "flowType": "FLOW_TYPE_LIVE_CHAT_ENGAGEMENT",
                            "flowEventMetadata": {
                              "liveChatEngagementContext": {
                                "entryPointType": "LIVE_CHAT_ENGAGEMENT_ENTRY_POINT_REPLY_BUTTON"
                              }
                            }
                          }
                        }
                      },
                      "loggingDirectives": {
                        "trackingParams": "MASKED_TRACKING_PARAMS",
                        "visibility": {
                          "types": "12"
                        }
                      }
                    }
                  },
                  "replyCountEntityKey": "MASKED_REPLY_COUNT_ENTITY_KEY",
                  "replyCountPlaceholder": {
                    "content": "Reply",
                    "styleRuns": [
                      {
                        "startIndex": 0,
                        "length": 5
                      }
                    ]
                  }
                }
              }
            }
          },
          "clientId": "MASKED_CLIENT_ID"
        }
      }
    ],
    "videoOffsetTimeMsec": "4150690",
    "isLive": true
  }
}
