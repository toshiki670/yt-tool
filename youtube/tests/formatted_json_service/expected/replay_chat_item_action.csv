id,authorExternalChannelId,postedAt,category,authorName,content
MASKED_ID,MASKED_CHANNEL_ID,2025-04-08T01:16:57.450013+09:00,Paid message,Masked Author Name,"message: 'Masked message content❗'; purchaseAmount: '¥1,000'; membershipMonths: 'Member (6 months)'"
